Add Lenovo ThinkSmart View (CD-18781Y)

kaechele opened the pull request:
This adds the device tree for the Lenovo ThinkSmart View (CD-18781Y).

Almost everything is working, except for:
- Proximity and Ambient Light Sensor (see warning in dts file)
- Camera (no driver available)
- Bluetooth Audio (lacking PCM support in upstream sound driver)

Sound support requires patches to the `tas5805m` ASoC codec driver that I'm intending to send upstream sometime soon.

barni2000 commented on arch/arm64/boot/dts/qcom/apq8053-lenovo-cd-18781y.dts:
```diff
@@ -0,0 +1,720 @@
+		drive-strength = <8>;
+	};
+
+	tlmm_sec_sus: sec-tlmm-off-pins  {
```
```suggestion
	tlmm_sec_sus: sec-tlmm-off-state  {
```

barni2000 commented:
Can you please squash device dt commits?
Do panel modifications are really needed?

kaechele commented:
Looks like the panel modifications weren't needed after all. Dropped them.

I squashed the device dts edits. I did leave the dtsi changes as a separate commit. Would you like me to squash those as well?

barni2000 commented:
> Looks like the panel modifications weren't needed after all. Dropped them.
> 
> I squashed the device dts edits. I did leave the dtsi changes as a separate commit. Would you like me to squash those as well?

No, dtsi commit should be sent upstream

barni2000 commented on arch/arm64/boot/dts/qcom/msm8953.dtsi:
```diff
@@ -2608,6 +2622,22 @@
 			qcom,controlled-remotely;
 		};
 
+		blsp2_uart0: serial@7aef000 {
```
blsp2_uart1 like the clock name GCC_BLSP2_UART1_APPS_CLK

barni2000 replied:
```

		uart_0: serial@78af000 {
			compatible = "qcom,msm-uartdm-v1.4", "qcom,msm-uartdm";
			reg = <0x078af000 0x200>;
			interrupts = <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>;
			clocks = <&gcc GCC_BLSP1_UART1_APPS_CLK>,
				 <&gcc GCC_BLSP1_AHB_CLK>;
			clock-names = "core", "iface";
			status = "disabled";
		};
```

barni2000 replied:
This also should called as blsp1_uart1 sadly it was named uart_0.

kaechele replied:
Good catch!

I adapted this from downstream: https://git.codelinaro.org/clo/la/kernel/msm-4.9/-/blob/LA.UM.10.6.2.c26/arch/arm64/boot/dts/qcom/msm8953.dtsi?ref_type=heads#L612

With a bit more experience now, I'm finding that `blsp2_uart0` should probably actually be `blsp_uart5`: https://git.codelinaro.org/clo/la/kernel/msm-4.9/-/blob/LA.UM.10.6.2.c26/arch/arm64/boot/dts/qcom/msm8953-pinctrl.dtsi#L608
This is also what the pins used downstream for `blsp2_uart0` are mapped to in [`drivers/pinctrl/qcom/pinctrl-msm8953.c`](https://github.com/msm8953-mainline/linux/blob/070c4019af1a4649f3268642c756f185fc5ecc8c/drivers/pinctrl/qcom/pinctrl-msm8953.c#L752) upstream.

I'll rewrite this commit using blsp_uart5 instead, as that is what MSM8916 and MSM8939 do upstream.
